fix: LEAP-2065: [FE] Remove warning noise from new Select component in Spark

nass600 opened the pull request:
Due to the recent refactor regarding the new Select component in our Design System (Spark), we are generating very noise warning messages when these components are present due to an uncontrolled/controlled hybrid implementation approach.

This ticket intends to remove these warnings

<img width="2541" alt="Screenshot 2025-04-30 at 16 38 30" src="https://github.com/user-attachments/assets/39705af2-97d5-4c6b-aa08-f3e0ffc14082" />

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nass600 committed:
fix: LEAP-2065: [FE] Remove warning noise from new Select component in Spark

yyassi-heartex commented:
/selenium test

nass600 commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14777968021)
> Successfully merged:  create mode 100644 web/libs/editor/tests/integration/e2e/view_all/readonly.cy.ts

nass600 committed:
Merge branch 'develop' into 'fb-leap-2065/remove-select-warns'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14777968021

hlomzik reviewed (changes requested):
The change looks kinda scary because it changes selected values of actual `select` tag. I'd ask for two things:
- check that it doesn't break forms relying on formdata especially on initial state without any changes from the user
- add a comment for the whole component describing the work with underlying hidden select, so we (and agents) can understand the principle and don't need to parse the code

hlomzik commented:
and definitely run selenium in LSE

nass600 commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14830785795)
> Successfully merged:  create mode 100644 web/tools/extract-antd-no-reset/package.json

nass600 committed:
Merge branch 'develop' into 'fb-leap-2065/remove-select-warns'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14830785795

nass600 commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14831659901)

nass600 committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14831659901

nass600 committed:
added a description header

nass600 commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14854046492)
> Successfully merged:  create mode 100644 .github/ISSUE_TEMPLATE/config.yml

nass600 committed:
Merge branch 'develop' into 'fb-leap-2065/remove-select-warns'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14854046492

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14854155497)

MihajloHoma committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14854155497
